packages/lit-element/src/lib/platform-support.ts

```diff
@@ -0,0 +1,413 @@
+     */
+    _createElement(html: string) {
+      const template = super._createElement(html);
+      if (!window.ShadyCSS!.nativeShadow) {
+        window.ShadyCSS!.prepareTemplateDom(template, currentScope);
+      }
+      if (!isScopeStyled(currentScope)) {
+        const scopeCss = cssForScope(currentScope);
+        // Remove styles and store textContent.
+        const styles = template.content.querySelectorAll('style') as NodeListOf<
+          HTMLStyleElement
+        >;
+        // Store the css in this template in the scope css.
+        scopeCss.push(
+          ...Array.from(styles).map((style) => {
+            style.parentNode?.removeChild(style);
+            return style.textContent!;
+          })
+        );
+      }
+      return template;
+    }
+  }
+
+  /**
+   * Patch to apply gathered css via ShadyCSS. This is done only once per scope.
+   */
+  NodePart.prototype.__baseSetValue = NodePart.prototype._setValue;
+  NodePart.prototype._setValue = function (
+    this: PatchableNodePart,
+    value: unknown
+  ) {
+    const container = this._startNode.parentNode!;
+    const renderingIntoShadowRoot = container instanceof ShadowRoot;
+    if (!renderingIntoShadowRoot) {
+      this.__baseSetValue(value);
+    } else {
+      const previousScope = currentScope;
+      currentScope = this.options.scope ?? '';
+
+      // Note, @apply requires outer => inner scope rendering on initial
+      // scope renders to apply property values correctly. Style preparation
+      // is tied to rendering into `shadowRoot`'s and this is typically done by
+      // custom elements. If this is done in `connectedCallback`, as is typical,
+      // the code below ensures the right order since content is rendered
+      // into a fragment first so the hosting element can prepare styles first.
+      // If rendering is done in the constructor, this won't work, but that's
+      // not supported in ShadyDOM anyway.
+      let renderContainer: DocumentFragment | undefined = undefined;
+      const startNode = this._startNode;
+      const endNode = this._endNode;
+      const needsStyleScoping = !isScopeStyled(currentScope);
+      if (needsStyleScoping && renderingIntoShadowRoot) {
+        renderContainer = document.createDocumentFragment();
+        renderContainer.appendChild(document.createComment(''));
+        // Temporarily move this part into the renderContainer.
+        this._startNode = this._endNode = renderContainer.firstChild!;
+      }
+      // Note, any nested template results render here and their styles will
+      // be extracted and collected.
+      this.__baseSetValue(value);
+
+      let template: HTMLTemplateElement | undefined = undefined;
+      if (needsStyleScoping && renderingIntoShadowRoot) {
+        // Get the template for this result or create a dummy one if a result
+        // is not being rendered.
+        template = !!(value as ShadyTemplateResult)?._$litType$
```

Comment:
To get the template, you could also do
```js
 template = this._value._template?._element || document.createElement('template')
```
after `__baseSetValue` since `__commitTemplateResult` sets the `part._value` to a `TemplateInstance` instance

Reply:
Done.